jobis-infrastructure/src/main/java/team/retum/jobis/domain/application/persistence/ApplicationPersistenceAdapter.java

```diff
@@ -376,4 +378,25 @@ private BooleanExpression recentRecruitment(Long companyId) {
     private BooleanExpression eqYear(Year year) {
         return year == null ? null : applicationEntity.createdAt.year().eq(year.getValue());
     }
+
+    @Override
+    public boolean existsAllByApplicationIdsAndCompanyId(List<Long> applicationIds, Long companyId) {
+        if (applicationIds == null || applicationIds.isEmpty()) {
+            return false;
+        }
+
+        Long count = queryFactory
+            .select(applicationEntity.count())
+            .from(applicationEntity)
+            .join(recruitmentEntity)
+            .on(applicationEntity.recruitment.id.eq(recruitmentEntity.id))
+            .where(
+                applicationEntity.id.in(applicationIds),
+                recruitmentEntity.company.id.eq(companyId)
+            )
+            .fetchOne();
+
+        Set<Long> uniqueIds = new HashSet<>(applicationIds);
+        return count != null && count == new HashSet<>(uniqueIds).size();
```

Comment:
HashSet을 중복 생성하지 말고, 아래와 같이 한 번만 생성하는 건 어떨까요?
```suggestion
        Set<Long> uniqueIds = new HashSet<>(applicationIds);
        return count != null && count == uniqueIds.size();
```